### PR TITLE
Improve message dialog buttons

### LINKED ISSUE #12027: translation: missing translation in lighttable delete selected images dialog
in lighttable delete selected images dialog the yes/no bottuons aren’t covered by translation:
image
**Platform**
* darktable version : darktable-3.9.0+1744~gebe766dba
* OS                : OSx

### PR DESCRIPTION
**Pull request by victoryforce**
This PR closes #12027 (and fixes not only the dialog specified there, but all similar ones) and also makes following improvements:
- buttons are guaranteed to be localized always and on all platforms
- buttons will get the same order on all platforms (previously there was a difference)
- finally there will be the same default action on all platforms (which is very important, because previously on Windows platform there was a confirmation action by default when asking to delete files - this is dangerous)

### CODE REVIEW

**Comment by dterrahe** [comment]
This introduces a bit of duplication. Could you refactor into a `dt_message_dialog_yes_no_new`? Easier to enforce consistency that way.

**Comment by victoryforce** [comment]
> This introduces a bit of duplication. Could you refactor into a `dt_message_dialog_yes_no_new`? Easier to enforce consistency that way.

@dterrahe  Yes, there will be _a bit of_ duplication. OK, code deduplication is almost always a good thing. And I can make a wrapper function (dt_message_dialog_yes_no_new), but not sure if it's worth it. I will try to explain.

The options (two of them) for creating such a function have their drawbacks, perhaps not very significant, but is the forced need to put up with them an adequate price for the ease of ensuring consistency for hypothetical cases when someone will add new similar dialogues in dt?

We will have to make a variadic wrapper of a variadic function. In the general case, this is impossible, such a possibility is not supported by the means of C language. Fortunately, it is possible to do this if the semantics of the tail of the arguments are identical to the semantics of the printf arguments (this is just our case). Thanks to the presence of vprintf-type functions, we can create a string with format specifiers already substituted, thus removing the need to pass a variable number of arguments to the wrapped function.

Nice idea, but we'll end up with a format string as a variable, not a string literal. Unfortunately, modern compilers issue a warning for this, since they cannot verify the correctness of the use of format substitutions (a format-security type warning).

This is fatal for the dt build because we build it with the "treat warnings as errors" option. We can get out of this situation by adding pragmas that will locally suppress this warning, but this starts to look like too ugly a solution (and we jeopardize support of compilers that are not pragma-compatible with the main ones, it's all very fragile).

Another implementation option is not to pass this text immediately to gtk_message_dialog_new, calling this function with an empty text. And then add text through one of the functions that allows you to construct a dialogue step by step, adding various properties to it, text, buttons, etc.

The disadvantage of this option is that the created function will not be a wrapper over `gtk_message_dialog_new`, but over a slightly different function `gtk_message_dialog_new_with_markup`. The reason is that there is a gtk_message_dialog_set_markup function in gtk, but there is not corresponding function that would add text to the dialog without markup. Not that it's a major drawback from a practical point of view, but still...

In conclusion, I can't choose which refactoring option I don't like less, and I can't convince myself that this idea is justified at all.

**Comment by dterrahe** [comment]
> This is fatal for the dt build

Oh no! This is very bad news indeed for `dt_print`, `dt_control_log`, `dt_util_dstrcat` that use this exact same approach and are used _all_ over the place! We'll need to find a replacement for them urgently!

**Comment by dterrahe** [comment]
These were included in #12911
